pkg/labels/array.go

```diff
@@ -91,17 +91,17 @@ nextLabel:
 // Implementation of the k8s.io/apimachinery/pkg/labels.Labels interface.
 func (ls LabelArray) Has(key string) bool {
 	// The key is submitted in the form of `source.key=value`
-	ck := GetCiliumKeyFrom(key)
-	keyLabel := ParseLabel(ck)
+	keyLabel := parseSelectLabel(key, '.')
 	if keyLabel.IsAnySource() {
-		for _, lsl := range ls {
-			if lsl.Key == keyLabel.Key {
+		for l := range ls {
+			if ls[l].Key == keyLabel.Key {
 				return true
 			}
 		}
 	} else {
 		for _, lsl := range ls {
-			if lsl.GetExtendedKey() == key {
+			// Note that if '=value' is part of 'key' it is ignored here
+			if lsl.Source == keyLabel.Source && lsl.Key == keyLabel.Key {
```

Comment:
@aanm Please have a look here. According to the comment above the key is of from `source.key=value`. If the `=value` part is really in there, then we used to never match that as `GetExtendedKey()` does NOT include the value in the returned string. It may be that the comment is wrong and the key never has the `=value` part, though.

Reply:
@jrajahalme where is this code now?

Reply:
@jrajahalme missing?

Reply:
This was merged already in #5967 that I had used as the basis for this PR. Please have look: https://github.com/cilium/cilium/pull/5967/files.